src/Database/PG/Query/Connection.hs

```diff
@@ -175,79 +178,93 @@ initPQConn ci logger =
 
     whenSerVerOk conn = do
       -- Set some parameters and check the response
-      mRes <- PQ.exec conn $ BL.toStrict $ BB.toLazyByteString $ mconcat
-              [ BB.string7 "SET client_encoding = 'UTF8';"
-              , BB.string7 "SET client_min_messages TO WARNING;"
-              ]
+      mRes <-
+        PQ.exec conn $
+          BL.toStrict $
+            BB.toLazyByteString $
+              mconcat
+                [ BB.string7 "SET client_encoding = 'UTF8';",
+                  BB.string7 "SET client_min_messages TO WARNING;"
+                ]
       case mRes of
         Just res -> do
           st <- PQ.resultStatus res
           case st of
             PQ.CommandOk -> return $ Right conn
-            _            -> return $ Left $
-                            PGConnErr "unexpected status after setting params"
-        Nothing  -> return $ Left $
-                    PGConnErr "unexpected result after setting params"
+            _ ->
+              return $
+                Left $
+                  PGConnErr "unexpected status after setting params"
+        Nothing ->
+          return $
+            Left $
+              PGConnErr "unexpected result after setting params"
 
 defaultConnInfo :: ConnInfo
 defaultConnInfo = ConnInfo 0 details
   where
-    details = CDOptions ConnOptions
-              { connHost = "127.0.0.1"
-              , connPort = 5432
-              , connUser = "postgres"
-              , connPassword = ""
-              , connDatabase = ""
-              , connOptions = Nothing
-              }
+    details =
+      CDOptions
+        ConnOptions
+          { connHost = "127.0.0.1",
+            connPort = 5432,
+            connUser = "postgres",
+            connPassword = "",
+            connDatabase = "",
+            connOptions = Nothing
+          }
 
 pgConnString :: ConnDetails -> DB.ByteString
 pgConnString (CDDatabaseURI uri) = uri
-pgConnString (CDOptions opts)    = fromString connstr
+pgConnString (CDOptions opts) = fromString connstr
   where
-    connstr = str "host="     connHost
-            $ num "port="     connPort
-            $ str "user="     connUser
-            $ str "password=" connPassword
-            $ str "dbname="   connDatabase
-            $ mStr "options=" connOptions []
+    connstr =
+      str "host=" connHost $
+        num "port=" connPort $
+          str "user=" connUser $
+            str "password=" connPassword $
+              str "dbname=" connDatabase $
+                mStr "options=" connOptions []
```

Comment:
Another case where repeated `$` application definitely makes things a bit weirder.

Reply:
Yeah, tempting to transform this to something like

```
compose
  [ str "user=" connUser,
    str "password=" connPassword,
    ...
  ]
  []
```

Reply:
Would `.` composition work here as well?

e.g. `str "users=" connUser . str "password=" connPassword . str "dbName=" connDatabase . mStr "options=" connOptions $ []`?

Reply:
I suppose the list construction/destruction shouldn't be an issue at all though :thinking:; I've just got a weirdly unhealthy aversion to Haskell lists I suppose.

Reply:
I think with `.` composition we'd still want to line-break the components, which I expect would staircase on that operator.

I don't intend to make any change here, though (more because `compose` doesn't exist rather than dislike of lists, but also this seems fine if a bit awkward).

Reply:
I was wrong here, `.` does indeed not staircase:

```
pgConnString (CDOptions opts) = fromString connstr
  where
    connstr =
      str "host=" connHost
        . num "port=" connPort
        . str "user=" connUser
        . str "password=" connPassword
        . str "dbname=" connDatabase
        . mStr "options=" connOptions
        $ []
```

(would still tend to leave as is)